src/views/ont/ONTRun.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <div id="ontrun">
+    <router-link :to="{ name: 'ONTRuns' }">
+      <traction-button id="backToRunsButton" class="float-right">Back</traction-button>
+    </router-link>
+
+    <traction-button
+      v-if="newRecord"
+      id="resetButton"
+      type="reset"
+      theme="default"
+      class="float-right"
+      @click="newRun"
+      >Reset</traction-button
+    >
+
+    <traction-button
+      :id="currentAction.id"
+      class="float-right"
+      :theme="currentAction.theme"
+      :disabled="!runValid"
+      @click="runAction"
+      >{{ currentAction.label }}</traction-button
+    >
+
+    <ONTRunInformation></ONTRunInformation>
+    <ONTRunInstrumentFlowcells></ONTRunInstrumentFlowcells>
+  </div>
+</template>
+<script>
+import ONTRunInformation from '@/components/ont/runs/ONTRunInformation'
+import ONTRunInstrumentFlowcells from '@/components/ont/runs/ONTRunInstrumentFlowcells'
+
+import { createNamespacedHelpers, mapActions } from 'vuex'
+const { mapGetters } = createNamespacedHelpers('traction/ont/runs')
+
+export default {
+  name: 'ONTRun',
+  components: {
+    ONTRunInformation,
+    ONTRunInstrumentFlowcells,
+  },
+  props: {
+    id: {
+      type: [String, Number],
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      newRecord: isNaN(this.id),
+      actions: {
+        create: {
+          id: 'create',
+          theme: 'create',
+          label: 'Create',
+          method: 'createRun',
+        },
+        update: {
+          id: 'update',
+          theme: 'update',
+          label: 'Update',
+          method: 'updateRun',
+        },
+      },
+    }
+  },
+  computed: {
+    ...mapGetters(['currentRun']),
+    currentAction() {
+      return this.actions[this.newRecord ? 'create' : 'update']
+    },
+    runValid() {
+      let flowcells = (this.currentRun.flowcell_attributes || []).filter(
+        (fc) => fc.flowcell_id && fc.tube_barcode,
+      )
+      return this.currentRun.instrument_name && this.currentRun.state && flowcells.length != 0
+    },
+  },
+  created() {
+    this.provider()
+  },
+  methods: {
+    ...mapActions('traction/ont/runs', [
+      'createRun',
+      'setInstruments',
+      'editRun',
+      'newRun',
+      'updateRun',
+    ]),
+    ...mapActions('traction/ont', ['fetchOntPools']),
+    async runAction() {
+      let response = await this[this.currentAction.method]()
+
+      if (response.success) {
+        this.redirectToRuns()
+      } else {
+        let action = this.newRecord ? 'create' : 'update'
+        this.showAlert(
+          `Failed to ${action} run in Traction: ${response.errors}`,
+          'danger',
+          'run-validation-message',
+        )
+      }
+    },
+    redirectToRuns() {
+      this.$router.push({ name: 'ONTRuns' })
+    },
+    async provider() {
+      await this.fetchOntPools()
```

Comment:
Can remove provider and created method by using datafetcher in the view to set the data

Reply:
@BenTopping